### ISSUE TITLE
referencesBatcher is cumbersome to use: new function suggested

### ISSUE THREAD

**Issue by erikvullings** [semi-technologies/weaviate-javascript-client#77]
The current API to update items in batch is rather cumbersome to use, e.g. according to the docs, I should use:
```js
const weaviate = require("weaviate-client");

const client = weaviate.client({
  scheme: 'http',
  host: 'localhost:8080',
});

client.batch
      .referencesBatcher()
      .withReference({
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4303/wroteArticles`,
        to: `weaviate://localhost/Article/6bb06a43-e7f0-393e-9ecf-3c0f4e129064`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      })
      .withReference({
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4303/wroteArticles`,
        to: `weaviate://localhost/Article/b72912b9-e5d7-304e-a654-66dc63c55b32`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      })
      .withReference({
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4304/wroteArticles`,
        to: `weaviate://localhost/Article/b72912b9-e5d7-304e-a654-66dc63c55b32`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      })
      .do()
      .then(res => {
        console.log(res)
      })
      .catch(err => {
          console.error(err)
      });
```
In my case, however, I do not know how many cross references I need to add (variable number), so it would be much easier and less verbose to use something as

```js
const weaviate = require("weaviate-client");

const client = weaviate.client({
  scheme: 'http',
  host: 'localhost:8080',
});

client.batch
      .referencesBatcher()
      .withReferences([{
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4303/wroteArticles`,
        to: `weaviate://localhost/Article/6bb06a43-e7f0-393e-9ecf-3c0f4e129064`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      }, {
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4303/wroteArticles`,
        to: `weaviate://localhost/Article/b72912b9-e5d7-304e-a654-66dc63c55b32`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      }, {
        from: `weaviate://localhost/Author/36ddd591-2dee-4e7e-a3cc-eb86d30a4304/wroteArticles`,
        to: `weaviate://localhost/Article/b72912b9-e5d7-304e-a654-66dc63c55b32`,
        // prior to v1.14 omit the class name as part of the "to" beacon and specify it as weaviate://localhost/<id>
      }])
      .do()
      .then(res => {
        console.log(res)
      })
      .catch(err => {
          console.error(err)
      });
```

I've briefly looked at the code, and it seems very easy to implement. Just add the following method to the [referencesBatcher.js](https://github.com/erikvullings/weaviate-javascript-client/blob/2f66b238568e1679ccdac10f1c1998310c131b89/batch/referencesBatcher.js#L14)

```js
withReferences = (obj) => {
  this.references = obj;
  return this;
};
```

**Comment by etiennedi** [comment]
@antas-marcin, @aliszka Any reason not to do this? From what I see it would be entirely non-breaking to add this alternative function? and just keep the old one?

**Comment by erikvullings** [comment]
@etiennedi Thanks for looking into it. And while you are considering it, you may actually add that code pattern to the `ObjectsBatcher` to, i.e.:

```js
 withObjects = (obj) => {
    this.objects = obj;
    return this;
  };
```

**Comment by aliszka** [comment]
`ReferencesBatcher::withReferences()` and `ObjectsBatcher::withObjects()` were added in https://github.com/semi-technologies/weaviate-javascript-client/pull/81 and will be available in the next release.

**Comment by aliszka** [comment]
Both methods available since release [v2.12.0](https://github.com/semi-technologies/weaviate-javascript-client/releases/tag/v2.12.0)
